server/src/main/java/org/cloudfoundry/identity/uaa/oauth/AntPathRedirectResolver.java

```diff
@@ -67,6 +69,45 @@ protected boolean redirectMatches(String requestedRedirect, String clientRedirec
         }
     }
 
+    /**
+     * Repeatedly:
+     * <ol>
+     *     <li>checks for a match</li>
+     *     <li>url decodes the requested path</li>
+     * </ol>
+     * until path cannot be url decoded any further. Then normalizes the path before the final check.
+     * <p>
+     *     For example, if example.com/foo is the registered url and example.com/foo/%252e./bar is the requested url,
+     *     checks a match for:
+     *     <ol>
+     *         <li>example.com/foo/%252e./bar</li>
+     *         <li>example.com/foo/%2e./bar</li>
+     *         <li>example.com/foo/../bar</li>
+     *         <li>example.com/bar</li>
+     *     </ol>
+     * </p>
+     */
+    private boolean matchesAfterNormalization(Predicate<String> matcher, String requestedRedirect) {
+        int maxDecodeAttempts = 5;
```

Comment:
Make this a Java constant please - at the start of the class:
```java
private static final int MAX_DECODE_ATTEMPTS = 5
```

Reply:
It is used in this method only. Why make it a field?